src/components/AttachmentPicker/index.native.tsx

```diff
@@ -158,12 +160,42 @@ function AttachmentPicker({type = CONST.ATTACHMENT_PICKER_TYPE.FILE, children, s
                         return reject(new Error(`Error during attachment selection: ${response.errorMessage}`));
                     }
 
-                    return resolve(response.assets);
+                    const targetAsset = response.assets?.[0];
+                    const targetAssetUri = targetAsset?.uri;
+
+                    if (!targetAssetUri) {
+                        return resolve();
+                    }
+
+                    if (targetAsset?.type?.startsWith('image')) {
+                        FileUtils.verifyFileFormat({fileUri: targetAssetUri, formatSignatures: CONST.HEIC_SIGNATURES})
+                            .then((isHEIC) => {
+                                // react-native-image-picker incorrectly changes file extension without transcoding the HEIC file, so we are doing it manually if we detect HEIC signature
```

Comment:
Do we plan to fix this upstream also?

Reply:
I've sent a feature request to the repository. If they think this approach is ok, I'll send a PR  https://github.com/react-native-image-picker/react-native-image-picker/issues/2317